README.rst

```diff
@@ -1,6 +1,24 @@
-==============
- REANA server
-==============
+======================
+REANA Server
+======================
```

Comment:
Any reason for having additional `=`? I remember that for the release checklist for Invenio we had to stick to the length of the tittle itself.